allennlp/training/trainer.py

```diff
@@ -117,10 +125,24 @@ def train(self) -> None:
                     loss = val_output_dict["loss"]
                     val_loss += loss.data.cpu().numpy()
 
-            # TODO(Mark): Add user specified metrics here, maybe a "metrics" key?
-            logger.info("Training Loss: %3f    Validation Loss: %3f ", train_loss, val_loss)
+                val_metrics = self._model.get_metrics(reset=True) or {}
```

Comment:
You could slightly simplify some of this logic by adding `metrics['Loss'] = train_loss; val_metrics['Loss'] = val_loss`.  Then you can actually make `self._validation_metric` default to `"loss"` like the docstring says it does (assuming you normalize capitalization issues), and not have any special casing for it below.